concepts/modularization/modularization_overview.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Modularization in Kyma
+---
+
+With Kyma’s modular approach, you can install exactly the Kyma components you want, making the installation very lightweight and adjusted to your business needs. Kyma provides three components to support the modular approach: The Module Manager, the Lifecycle Manager, and the Runtime Watcher.
+
+## Kyma’s Modular Approach
+
+![](assets/Modular_Approach.svg)
+
+## 1. Kyma Module
+
+A Kyma module contains all information required to install and run the associated components in a Kyma runtime:
+
+* A renderable Kubernetes manifest that includes the Kubernetes resources of the module
+* A default configuration for module chart resource installation
+* Optionally, further information
+
+All these assets are bundled into a single container image using the OCI image specification.
+
+## 2. Module Manager
+
+The Module Manager installs and manages the Kyma components in a local or remote setup. For example, it remotely manages and synchronizes the installation in the SAP Kyma Runtime. The Module Manager retrieves the container image and processes the Kubernetes manifest by calling a rendering framework like Helm or Kustomize. Finally, it deploys the manifest in Kubernetes and monitors the outcome of the deployment.
+
+## 3. Runtime Watcher
+
+The Runtime Watcher monitors the relevant Kyma resources of the user’s Kyma runtime for any changes, such as:
+
+* Expected changes, like a Kyma user providing a configuration update, which must lead to an update of the particular Kyma runtime.
+* Unexpected changes, like an accidental deletion or modification, which must be reverted to recover the Kyma runtime to a healthy state.
+
+If needed, the Runtime Watcher triggers a reconciliation of the user’s Kyma runtime by propagating the desired changes to the Module Manager and the Lifecycle Manager.
+
+## 4. Lifecycle Manager
+
+The Lifecycle Manager is responsible for managing the lifecycle of Kyma instances, like installing or deleting a Kyma instance from a cluster. It checks whether a CR is provided for every module, and deletes obsoletes module if they’re no longer required
```

Comment:
```suggestion
The Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. This is done by generating required Kyma module custom resources, which bootstrap Kyma module processing. To add, it reconciles a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster. 
```

Reply:
Slight editing for @adityabhatia's suggestion

```suggestion
The Lifecycle Manager is responsible for orchestrating Kyma module operators to process their respective resources on the Kyma runtime. This is done by generating the required Kyma module custom resources, which bootstrap Kyma module processing. Furthermore, the Lifecycle Manager reconciles a Kyma custom resource for each Kyma runtime, indicating the consolidated status of all modules configured for that Kyma cluster. 
```

Reply:
@adityabhatia, what **_are_** module operators? It's the first time we mention the term here. We need to explain it in this overview doc.

Reply:
Module operators are basically component operators, which will be installed by other Kyma component teams or third party component owners. These operators are responsible for managing the module (component) specific resources on SKR

Reply:
Because you wrote "basically" - are there other kinds of component operators? Or is "module operator" synonym to "component operator"?

Reply:
Yes, we decided to call it module instead of a component, that is the reason we call it a module-operator

Reply:
Module Operators are Controllers that steer the lifecycle of a single Module. A Module can consist of one or many compoents.

Example:
Telemetry Operator (Operator) is in charge of reconciling LogPipeline and other CRs. It controls the Kyma Components "Logging", "Monitoring" and "Tracing".

Reply:
Thank you! Where would module operators show up in the diagram? Where do they belong in the text?

Reply:
They are abstracted under "Kyma Component". Technically it can also be a pure helm chart without an operator, but this is not recommended by the architecture design right now.